main/lsp/ErrorReporter.h

```diff
@@ -27,6 +36,10 @@ class ErrorReporter {
      */
     void pushDiagnostics(u4 epoch, core::FileRef file, const std::vector<std::unique_ptr<core::Error>> &errors,
                          const core::GlobalState &gs);
+
+    void beginEpoch(u4 epoch, std::vector<std::unique_ptr<Timer>> diagnosticLatencyTimers);
+    void endEpoch(u4 epoch);
```

Comment:
WDYT about making this `void endEpoch(u4 epoch, bool committed)` and removing `cancelEpoch`?